src/TrackGenerator.cpp

```diff
@@ -559,7 +605,9 @@ void TrackGenerator::generateTracks() {
     }
   }
 
+  /* Initialize the track boundary conditions and set the track UIDs */
   initializeBoundaryConditions();
+  initializeTrackPeriodicCycleIndices();
```

Comment:
In 3D MOC we have both periodic and reflective cycles, so I decided to use the "Periodic" specifier here.